zebra-state/src/constants.rs

```diff
@@ -117,6 +117,12 @@ pub const MAX_FIND_BLOCK_HEADERS_RESULTS: u32 = 160;
 /// These database versions can be recreated from their directly preceding versions.
 pub const RESTORABLE_DB_VERSIONS: [u64; 1] = [26];
 
+/// The maximum number of invalidated blocks per entry.
+///
+/// This limits the memory for each entry to around:
+/// `256 blocks * 2 MB per block = 0.5 GB`
+pub const MAX_INVALIDATED_BLOCKS: usize = 256;
```

Comment:
If the collection could be mutated by peers, it would be better to track the size of the blocks and start dropping the first ones that were inserted until it's below a size limit, but limiting the number of entries is good enough here as invalidated blocks should only be added via a private RPC interface.

```suggestion
/// The maximum number of invalidated block records.
///
/// This limits the memory use to around:
/// `100 entries * up to 99 blocks * 2 MB per block = 20 GB`
pub const MAX_INVALIDATED_BLOCKS: usize = 100;
```